### PR TITLE
PSP34 controller + wrapper

### PR DESCRIPTION
**Pull request by PierreOssun**
- Added PSP34 controller + wrapper
- updated Readme accordingly

### CODE REVIEW

File: xvm-sdk-contracts/psp34-controller/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ink_storage = { version = "3.4.0", default-features = false }
+ink_lang = { version = "3.4.0", default-features = false }
+ink_prelude = { version = "3.4.0", default-features = false }
+ethabi = { git = "https://github.com/akru/ethabi", default-features = false }
+hex-literal = "0.3"
+
+scale = { package = "parity-scale-codec", version = "3", default-features = false, features = ["derive"] }
+scale-info = { version = "2", default-features = false, features = ["derive"], optional = true }
+
+xvm-sdk = { path = "../../lib/xvm-environment", default-features = false }
```
Comment (akru) [inline xvm-sdk-contracts/psp34-controller/Cargo.toml:20]: I believe it’s not intuitive import package with one name from directory with different name.

File: xvm-sdk-contracts/psp34-controller/lib.rs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use self::psp34::{
+    PSP34Controller,
+    PSP34ControllerRef,
+};
+use ink_lang as ink;
+
+/// EVM ID (from astar runtime)
+const EVM_ID: u8 = 0x0F;
```
Comment (akru) [inline xvm-sdk-contracts/psp34-controller/lib.rs:11]: Let’s make env constant.

File: xvm-sdk-contracts/psp34-controller/lib.rs [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+                    Vec::from(self.evm_address.as_ref()),
+                    encoded_input,
+                )
+                .map_err(|_| PSP34Error::Custom(String::from("approve failed")))
+        }
+
+        #[ink(message, selector = 0x3128d61b)]
+        pub fn transfer(&mut self, to: AccountId, id: Id, _data: Vec<u8>) -> Result<(), PSP34Error> {
+            let caller = self.env().caller();
+            let encoded_input = Self::transfer_from_encode(Self::h160(&caller), Self::h160(&to), id.into());
```
Comment (akru) [inline xvm-sdk-contracts/psp34-controller/lib.rs:82]: So, here we expect that caller approve some tokens to contract first?

Comment (PierreOssun) [reply xvm-sdk-contracts/psp34-controller/lib.rs:82]: in PSP34 there is only `transfer` not `transfer_from`
in ERC721 there is only `transfer_from`
So in PSP34 `transfer` I should use ERC721 `transfer_from`

[as caller == from no need allowance](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/b2970b96e5e2be297421cd7690e3502e49f7deff/contracts/token/ERC721/ERC721.sol#L156)

File: xvm-sdk-contracts/psp34-controller/lib.rs (hunk 2 above)
Comment (PierreOssun) [reply xvm-sdk-contracts/psp34-controller/lib.rs:11]: What do you mean exactly?

**Comment by akru** [comment]
Please fix merge collisions and lest get it merged

**Pull request merged**